plugin/environment.go

```diff
@@ -95,14 +97,34 @@ func (env *Environment) Active() []*model.BundleInfo {
 
 // IsActive returns true if the plugin with the given id is active.
 func (env *Environment) IsActive(id string) bool {
-	_, ok := env.activePlugins.Load(id)
-	return ok
+	return env.GetPluginState(id) == model.PluginStateRunning
+}
+
+// GetPluginState returns the current state of a plugin (disabled, running, or error)
+func (env *Environment) GetPluginState(id string) int {
+	rp, ok := env.registeredPlugins.Load(id)
+	if !ok {
+		return model.PluginStateNotRunning
+	}
+
+	return *rp.(*registeredPlugin).State
+}
+
+// SetPluginState sets the current state of a plugin (disabled, running, or error)
+func (env *Environment) SetPluginState(id string, state int) bool {
```

Comment:
Maybe (1/5) change the return value to `error`? Trying to set the state that is not running, seams to be an error case to me.

Reply:
If a plugin does not exist in this map at the time of calling this function, it is because the plugin is installed and configured as disabled on the server, and has not been enabled/activated in the current server run.

The case where this call would be made is when the plugin is being marked as disabled in `app/plugin.go` line 76, because of a config change sync (not sure how to permalink a line in current PR changes). This is a redundant call because it is already disabled. So as the code currently stands, it is not an error if the plugin is not in the map. I believe another way of handling this would be to add the plugin to the map during `SetPluginState`, if it does not currently exist, then have a void return value.

Another route entirely would be to make it so the map always reflects all available/installed plugins. This makes me concerned that other packages need to be aware of calling say a new function `env.InstallPlugin` when a new plugin is added. Though, this solution clears up the ambiguity of the `registeredPlugins` (it could then be renamed to `installedPlugins`), and this would make it so returning an error would be appropriate if it does not exist in the map. On server startup, the `installedPlugins` map would be populated with all that are returned from `env.Available()`. Managing the sync between the installed plugins folder and the map is something I wanted to avoid, but it makes for better error handling.

@hanzei What are your thoughts?

Reply:
Maybe since this is considered an operation that can't fail, we can remove the return value?